### ISSUE TITLE
Generic NetworkOperations

### ISSUE THREAD

**Issue by josefdolezal** [jakub-tucek/fit-checker#23]
Before we start implementing new features (such as KOS, Obedar/LunchGuy, ..) we should refactor network operations. Now there are 3 operations (2 html, 1 json).

Each operation now has similar structure to this:
```swift
init(someArgs:) {}

override func start() {}

func handle(response:) {}
```

Lets make some generic abstraction over these operation. My suggestion is:

0. Make custom serializer for Realm objects [docs](https://github.com/Alamofire/Alamofire#creating-a-custom-response-serializer)
1. Add generic methods `success` and `failure`, where `failure` may have default implementation (in protocol or parent class)

We can discuss it here and then split it into multiple issues.
